controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java

```diff
@@ -116,7 +116,11 @@ private void cancelUpgradesOf(ApplicationList applications, String reason) {
 
     /** Returns the number of applications to upgrade in this run */
     private int numberOfApplicationsToUpgrade() {
-        return Math.max(1, (int) (maintenanceInterval().getSeconds() * (upgradesPerMinute() / 60)));
+        long intervalMillis = maintenanceInterval().dividedBy(Math.max(1, controller().curator().cluster().size())).toMillis();
+        long intervalStart = ((controller().clock().millis() + intervalMillis / 2) / intervalMillis) * intervalMillis;
+
+        long upgradeIntervalMillis = upgradesPerMinute() > 60e3 ? 1 : upgradesPerMinute() < 60e-3 ? 1_000_000 : (long) (60_000 / upgradesPerMinute());
```

Comment:
Nested ternary 😞 

Also, extracting variables for the magic constants would make this more readable.  E.g. `oneMinute` instead of `60e3` / `60_000`.

Reply:
I'd argue it's rather a chained ternary, which reads pretty well as a switch without fallthrough, and with a default clause at the end :) 

In any case, I managed to avoid it completely, by not replacing // with * :)